### PR TITLE
print_tree UPat +fix

### PR DESCRIPTION
**Pull request by DKormann**
print Upat and let const 0 uops be displayed correctly

### CODE REVIEW

File: tinygrad/engine/graph.py [hunk 1]
```diff
@@ -77,12 +77,12 @@ def log_lazybuffer(lb:'LazyBuffer', scheduled=False):
 
 def _tree(luop:Union[LazyOp,UOp], cycles, cnt, prefix=""):
```
Comment (Qazalin) [inline tinygrad/engine/graph.py:78]: function signature isn't changed.

File: tinygrad/engine/graph.py [hunk 2]
```diff
@@ -75,21 +75,22 @@ def log_lazybuffer(lb:'LazyBuffer', scheduled=False):
...
+  childs = [_tree(c, cycles, cnt) for c in src]
   for c in childs[:-1]: lines += [f" ┣{c[0]}"] + [f" ┃{l}" for l in c[1:]]
   return lines + [" ┗"+childs[-1][0]] + ["  "+l for l in childs[-1][1:]]
 
-def print_tree(luop:Union[LazyOp,UOp]): print("\n".join([f"{str(i).rjust(3)} {s}" for i,s in enumerate(_tree(luop, {}, [-1]))]))
+def print_tree(dag:Union[LazyOp, UOp, UPat]): print("\n".join([f"{str(i).rjust(3)} {s}" for i,s in enumerate(_tree(dag, {}, [-1]))]))
 
 def graph_uops(uops:List[UOp]):
-  colors = {UOps.ALU: "#ffffc0", UOps.LOAD: "#ffc0c0", UOps.STORE: "#c0ffc0", UOps.SPECIAL: "#c0c0ff", UOps.CONST: "#e0e0e0",
+  colors = {UOps.ALU: "#ffffc0", UOps.LOAD: "#ffc0c0", UOps.STORE: "#c0ffcs0", UOps.SPECIAL: "#c0c0ff", UOps.CONST: "#e0e0e0",
```
Comment (geohot) [inline tinygrad/engine/graph.py:93]: There is a typo here. Can you add tests in CI for this?

**Comment by geohot** [comment]
Can you add tests to CI for print_tree that no behavior regressed?

**Comment by DKormann** [comment]
fyi had slight behaviour changes
for uop const zero is printing out as CONST 0
for LazyOp ADD is printing as BinaryOps.ADD

**Pull request merged**